### DISCUSSION TITLE
TypeError: events.some is not a function

### DISCUSSION THREAD

**Discussion by hari5udhan** [serverless/serverless#11540] [Q&A]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

**This the file in which it shows error and I have Highlighted that code also**

'use strict';

const _ = require('lodash');

class AwsCompileAlexaSkillEvents {
  constructor(serverless) {
    this.serverless = serverless;
    this.provider = this.serverless.getProvider('aws');

    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'alexaSkill', {
      anyOf: [
        { $ref: '#/definitions/awsAlexaEventToken' },
        {
          type: 'object',
          properties: {
            appId: { $ref: '#/definitions/awsAlexaEventToken' },
            enabled: { type: 'boolean' },
          },
          required: ['appId'],
          additionalProperties: false,
        },
      ],
    });

    this.hooks = {
      'initialize': () => {
        if (
          this.serverless.service.provider.name === 'aws' &&
          Object.values(this.serverless.service.functions).some(({ events }) =>
            **_events.some((event) => event === 'alexaSkill')_**
          )
        ) {
          this.serverless._logDeprecation(
            'ALEXA_SKILL_EVENT_WITHOUT_APP_ID',
            'Starting with next major version, support for alexaSkill event without appId specified will be removed.'
          );
        }
      },
      'package:compileEvents': this.compileAlexaSkillEvents.bind(this),
    };
  }

  compileAlexaSkillEvents() {
    this.serverless.service.getAllFunctions().forEach((functionName) => {
      const functionObj = this.serverless.service.getFunction(functionName);
      let alexaSkillNumberInFunction = 0;

      functionObj.events.forEach((event) => {
        if (event.alexaSkill) {
          let enabled = true;
          let appId;
          if (typeof event.alexaSkill === 'string') {
            appId = event.alexaSkill;
          } else {
            appId = event.alexaSkill.appId;
            // Parameter `enabled` is optional, hence the explicit non-equal check for false.
            enabled = event.alexaSkill.enabled !== false;
          }
          alexaSkillNumberInFunction++;

          const lambdaLogicalId = this.provider.naming.getLambdaLogicalId(functionName);

          const permissionTemplate = {
            Type: 'AWS::Lambda::Permission',
            Properties: {
              FunctionName: {
                'Fn::GetAtt': [lambdaLogicalId, 'Arn'],
              },
              Action: enabled ? 'lambda:InvokeFunction' : 'lambda:DisableInvokeFunction',
              Principal: 'alexa-appkit.amazon.com',
            },
          };

          if (appId) {
            permissionTemplate.Properties.EventSourceToken = appId.replace(/\\n|\\r/g, '');
          }

          const lambdaPermissionLogicalId =
            this.provider.naming.getLambdaAlexaSkillPermissionLogicalId(
              functionName,
              alexaSkillNumberInFunction
            );

          const permissionCloudForamtionResource = {
            [lambdaPermissionLogicalId]: permissionTemplate,
          };

          _.merge(
            this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
            permissionCloudForamtionResource
          );
        }
      });
    });
  }
}

module.exports = AwsCompileAlexaSkillEvents;

### Service configuration (serverless.yml) content

```yaml
service: certificate-api

frameworkVersion: '3'

provider:
  name: aws
  runtime: nodejs12.x


functions:
  hello:
    handler: handler.createCertificate
    events:
      -http: 
        method: post
        path: certificationList
```

### Command name and used flags

When I run serveless deploy I get this error!

### Command output

```shell
Error:
TypeError: events.some is not a function
    at /usr/local/lib/node_modules/serverless/lib/plugins/aws/package/compile/events/alexa-skill.js:30:20
    at Array.some (<anonymous>)
    at initialize (/usr/local/lib/node_modules/serverless/lib/plugins/aws/package/compile/events/alexa-skill.js:29:60)
    at PluginManager.run (/usr/local/lib/node_modules/serverless/lib/classes/plugin-manager.js:598:65)
    at async Serverless.run (/usr/local/lib/node_modules/serverless/lib/serverless.js:170:5)
    at async /usr/local/lib/node_modules/serverless/scripts/serverless.js:787:9
```

### Environment information

```shell
serverless version
Framework Core: 3.25.0
Plugin: 6.2.2
SDK: 4.3.2

node version
v18.12.1
```

**Comment by medikoo** [comment]
@hari5udhan add `configValidationMode: error` to your config, it'll properly expose a configuration error (which you anyway have reported as a warning). With configuration errors any dirty crash is possible

**Comment by hari5udhan** [comment]
> configValidationMode

could you please elaborate the command in syntax

**Comment by medikoo** [comment]
> could you please elaborate the command in syntax

It's a configuration property you should add to `serverless.yml`

**Comment by hari5udhan** [comment]
service: certificate-api

frameworkVersion: '3'
configValidationMode: error

provider:
  name: aws
  runtime: nodejs12.x
  


functions:
  hello:
    handler: handler.createCertificate
    events:
      -http: 
        method: post
        path: certificationList


# you can add CloudFormation resource templates here
#resources:
#  Resources:
#    NewResource:
#      Type: AWS::S3::Bucket
#      Properties:
#        BucketName: my-new-bucket
#  Outputs:
#     NewOutput:
#       Description: "Description for the output"
#       Value: "Some output value"

I have added **configValidationMode: error**
now it shows like this
but I haven't used functions.hello.events
My handler is:
'use strict';

module.export.createCertificate = async (event) => {
  return {
    statusCode: 200,
    body: JSON.stringify("Certification added!"),
  };
};

Error:
Configuration error at 'functions.hello.events': must be array

**Comment by medikoo** [comment]
Error message clearly states where there's a configuration in your `serverless.yml`. Once you address it, it'll start to work

**Comment by medikoo** [comment]
@hari5udhan to see it better you may use https://jsonformatter.org/yaml-to-json (as by looking at YAML it may not be that obvious)

**Reply by hari5udhan** [reply]
Thankyou
